src/components/CytoscapeGraph/CytoscapeReactWrapper.tsx

```diff
@@ -52,32 +56,41 @@ export class CytoscapeReactWrapper extends React.Component<CytoscapeReactWrapper
   cy: any;
   divParentRef: any;
 
+  private static getUrlNodeType(data: any) {
+    const nodeType = data.nodeType;
+    switch (nodeType) {
+      case 'app':
+        return 'applications';
+      case 'service':
+        return 'services';
+      case 'workload':
+        return 'workloads';
+      default:
+        return '';
+    }
+  }
+
   // @todo: We need take care of this at global app level
   private static makeDetailsPageUrl(element: any) {
     const data = element.data();
     const namespace = data.namespace;
-    const nodeType = data.nodeType;
-    const workload = data.workload;
-    let app = data.app;
-    let urlNodeType = app;
-    if (nodeType === 'app') {
-      urlNodeType = 'applications';
-    } else if (nodeType === 'service') {
-      urlNodeType = 'services';
-    } else if (workload) {
-      urlNodeType = 'workloads';
-      app = workload;
-    }
-    return `/namespaces/${namespace}/${urlNodeType}/${app}`;
+    const app = data.nodeType === 'workload' ? data.workload : data.app;
+
+    return `/namespaces/${namespace}/${this.getUrlNodeType(data)}/${app}`;
   }
 
   private static redirectContextLink(e: MouseEvent) {
     e.preventDefault();
     if (e.target) {
       const anchor = e.target as HTMLAnchorElement;
       const href = anchor.getAttribute('href');
-      if (href) {
+      const newTab = anchor.getAttribute('target') === '_blank';
+      if (href && !newTab) {
         history.push(href);
+      } else {
+        if (href) {
+          window.open(href, '_blank');
```

Comment:
If  you are going to open a new tab, I think it is probably better to not use Javascript/Typescript to open the tab. Probably, it is better to not call `e.preventDefault()` and just let the browser to handle the redirection.

If  you open a new window through JS/TS, sometimes you may get that annoying warning about that the page is trying to open a new window, and the browser may prevent the window from opening until the user accepts opening the window.

Reply:
We can't do it like is the code now with the a element, It's ok if we are pushing the story but not a new tab, maybe after KIALI-2919

Reply:
I was meaning to do the exception only when target=_blank, just let the browser handle the click. For the other cases, it is OK to handle with JS/TS.

Reply:
but @israel-hdez there is a control there when is "_blank"  with the variable newTab

```typescript
private static redirectContextLink(e: MouseEvent) {
    e.preventDefault();
    if (e.target) {
      const anchor = e.target as HTMLAnchorElement;
      const href = anchor.getAttribute('href');
      const newTab = anchor.getAttribute('target') === '_blank';
      if (href && !newTab) {
        history.push(href);
      } else {
        if (href) {
          window.open(href, '_blank');
        }
      }
    }
  }
```